### ISSUE TITLE
Could not find class 'android.graphics.drawable.RippleDrawable'

### ISSUE THREAD

**Issue by leijdekkers** [Clans/FloatingActionButton#130]
Maybe nothing major but i see these errors thrown when i have a layout with the FAB button. Any idea what goes wrong?

Could not find class 'android.graphics.drawable.RippleDrawable', referenced from method com.github.clans.fab.FloatingActionButton.createFillDrawable

Could not find class 'android.graphics.drawable.RippleDrawable', referenced from method com.github.clans.fab.FloatingActionButton.onActionDown

 Could not find class 'android.graphics.drawable.RippleDrawable', referenced from method com.github.clans.fab.FloatingActionButton.onActionUp

**Comment by lolkittens** [comment]
What's your targetSdkVersion?

**Comment by leijdekkers** [comment]
Hi

The tarketSDK is 23

Any idea what it could be?

Peter

**Comment by lolkittens** [comment]
I didn't realize these were runtime errors, these can be safely ignored. It may be worth the effort to prevent Api 21+ methods from being called on older Android versions.

**Comment by leijdekkers** [comment]
Indeed. But i assume the library should check this and not me in my own code?

**Comment by lolkittens** [comment]
Yes, it would need to be handled internally.

**Comment by whyte624** [comment]
:+1: waiting for new release without this error.

**Comment by ckarthikv** [comment]
Has this error been sorted in the latest build? I keep getting crashes for SDK19 using this.

**Comment by CSMahajan** [comment]
I have the same errors , does anyone the solution for it?I tried for both targetSdkVersion 23 and my device api that is prelollipop but can't resolve it.

**Comment by jfest** [comment]
Hi, 
still having this error with device api 18,

how to fix it plz ?

**Comment by jfest** [comment]
Just include : compile 'com.android.support:support-annotations:XXXX'
Solve the problem, it was fix in code with annotation @Target...

**Comment by benidict1995** [comment]
anyone try to code FAB inside service class? .. because i want FAB move around the screen and also i can multi task.

example: while FAB on my screen, i can use any app i can tap what i want.

actually i've done this before but in simple button not FAB.

**Comment by monemihir** [comment]
i've just seen this error

```
Could not find class 'android.graphics.drawable.RippleDrawable', referenced from method com.github.clans.fab.FloatingActionButton.createFillDrawable
```

this causes the layout inflating to fail since the ripple effect is specified in the XML.

i'm targeting SDK15, device is running SDK17 and building with SDK25.
